api/src/main/java/com/messagebird/MessageBirdServiceImpl.java

```diff
@@ -253,6 +274,63 @@ public void deleteByID(String request, String id) throws UnauthorizedException,
         }
     }
 
+    /**
+     *
+     * Do get request for file from input url and stores the file in filepath.
+     * @param url     Absolute URL.
+     * @param filePath the path where the downloaded file is going to be stored.
+     * @return if it succeed, it returns filepath otherwise null or exception.
+     */
+    String doGetRequestForFileAndStore(final String url, final String filePath) throws GeneralException, UnauthorizedException, NotFoundException {
+        HttpURLConnection connection = null;
+        InputStream inputStream = null;
+
+        try {
+            connection = getConnection(url, null, METHOD_GET);
+            int status = connection.getResponseCode();
+
+            if (APIResponse.isSuccessStatus(status)) {
+                inputStream = connection.getInputStream();
+            } else {
+                inputStream = connection.getErrorStream();
+            }
+            if (status == HttpURLConnection.HTTP_OK) {
+                return writeInputStreamToFile(inputStream, filePath);
+            }
+            String body = readToEnd(inputStream);
+            handleHttpFailStatuses(status, body);
+        } catch (IOException ioe) {
+            throw new GeneralException(ioe);
+        } finally {
+            saveClose(inputStream);
+            if (connection != null) {
+                connection.disconnect();
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Writes input stream from IO to filepath.
+     * @param inputStream stream that has been collected file input
+     * @param filepath the storage path for the file
+     * @return if it succeed, it returns filepath otherwise null or exception.
+     * @throws IOException
+     */
+    private String writeInputStreamToFile(InputStream inputStream, String filepath) throws IOException {
+        // opens an output stream to save into file
+        FileOutputStream outputStream = new FileOutputStream(filepath);
+
+        int bytesRead = -1;
+        byte[] buffer = new byte[BUFFER_SIZE];
+        while ((bytesRead = inputStream.read(buffer)) != -1) {
```

Comment:
I couldn't get logic exactly in while, can you explain a bit.

Reply:
It reads 4096 byte each time in the while loop until it reaches EOF instead of reading the whole file at one time. Also, in each time in the loop, it writes the bytes to a file instead of keeping them in memory. http://tutorials.jenkov.com/java-io/inputstream.html .